spark-cassandra-connector/src/main/scala/org/apache/spark/sql/cassandra/CassandraCatalog.scala

```diff
@@ -1,65 +1,242 @@
+  def getClusters(): Seq[String] = synchronized {
+    metaStore.getAllClusters()
+  }
+
+  /** Create a database in metastore */
+  def createDatabase(database: String, cluster: Option[String]): Unit = synchronized {
+    metaStore.storeDatabase(database, cluster)
+  }
+
+  /** Create a cluster in metastore */
+  def createCluster(cluster: String): Unit = synchronized {
+    metaStore.storeCluster(cluster)
+  }
+
+  /** Refresh CassandraContext schema cache, then refresh table in local cache */
+  def refreshTable(tableRef: TableRef): Unit = {
+    cc.refreshCassandraSchema(tableRef.cluster.getOrElse(cc.getCluster))
+    cachedDataSourceTables.refresh(catalystTableIdentFrom(tableRef))
+  }
+
+  /** Refresh CassandraContext schema cache, then refresh table in local cache */
+  override def refreshTable(databaseName: String, tableName: String): Unit = {
+    refreshTable(TableRef(tableName, databaseName, Option(cc.getCluster)))
+  }
+
+  /** Return table metadata */
+  def getTableMetadata(tableRef : TableRef) : Option[TableMetaData] = synchronized {
+    metaStore.getTableMetaData(tableRef)
   }
 
-  override def registerTable(tableIdentifier: Seq[String], plan: LogicalPlan): Unit = ???
+  /** Set table schema */
+  def setTableSchema(tableRef : TableRef, schemaJsonString: String) : Unit = {
+    synchronized {
+      metaStore.setTableSchema(tableRef, schemaJsonString)
+    }
+    cachedDataSourceTables.refresh(catalystTableIdentFrom(tableRef))
+  }
 
-  override def unregisterTable(tableIdentifier: Seq[String]): Unit = ???
+  /** Set an option of table options*/
+  def setTableOption(tableRef : TableRef, key: String, value: String) : Unit = {
+    synchronized {
+      metaStore.setTableOption(tableRef, key, value)
+    }
+    cachedDataSourceTables.refresh(catalystTableIdentFrom(tableRef))
+  }
 
-  override def unregisterAllTables(): Unit = ???
+  /** Remove an option from table options */
+  def removeTableOption(tableRef : TableRef, key: String) : Unit = {
+    synchronized {
+      metaStore.removeTableOption(tableRef, key)
+    }
+    cachedDataSourceTables.refresh(catalystTableIdentFrom(tableRef))
+  }
 
-  override def tableExists(tableIdentifier: Seq[String]): Boolean = {
-    val (cluster, database, table) = getClusterDBTableNames(tableIdentifier)
-    val schema = schemas.get(cluster)
-    val tabDef = for (ksDef <- schema.keyspaceByName.get(database); tabDef <- ksDef.tableByName.get(table)) yield tabDef
-    tabDef.isDefined
+  /** Remove table schema */
+  def removeTableSchema(tableRef : TableRef) : Unit = {
+    synchronized {
+      metaStore.removeTableSchema(tableRef)
+    }
+    cachedDataSourceTables.refresh(catalystTableIdentFrom(tableRef))
   }
 
-  override def getTables(databaseName: Option[String]): Seq[(String, Boolean)] = ???
+  /** Convert Catalyst tableIdentifier to TableRef */
+  def tableRefFrom(tableIdentifier: Seq[String]) : TableRef = {
+    val id = processTableIdentifier(tableIdentifier).reverse.lift
```

Comment:
Can you elaborate it more. The method converts Seq[String] to TableRef(table, keyspace, cluster).